src/endpoints/authentication/impl.cpp

```diff
@@ -20,120 +21,484 @@
+				const auto iter{hdrs.find("authorization")};
+
+				// Failed to find auth header, use OIDC auth_code flow instead
+				if (iter == std::end(hdrs)) {
+					return _sess_ptr->send(fail(status_type::bad_request));
+				}
+
+				log::debug("{}: Authorization value: [{}]", fn, iter->value());
+
+				//
+				// TODO Here is where we determine what form of authentication to perform (e.g. Basic or OIDC).
+
+				//
+				// BLAH BLAH BLAH, assume we have the user & pass...
+				// Prob via Proxy-Authorization????
+				//
+
+				const auto pos = iter->value().find("Basic ");
+				if (std::string_view::npos == pos) {
+					// TEMPORARY PLS MAKE BETTER LATER :)
+
+					const auto alt_method{iter->value().find("iRODS ")};
+					if (std::string_view::npos == alt_method) {
+						return _sess_ptr->send(fail(status_type::bad_request));
+					}
+
+					// BEGIN BASE64 HEADER DECODE
+
+					constexpr auto basic_auth_scheme_prefix_size = 6;
+					std::string authorization{iter->value().substr(alt_method + basic_auth_scheme_prefix_size)};
+					boost::trim(authorization);
+					log::debug("{}: Authorization value (trimmed): [{}]", fn, authorization);
+
+					constexpr auto max_creds_size = 128;
+					unsigned long size = max_creds_size;
+					//std::vector<std::uint8_t> creds(size);
+					std::array<std::uint8_t, max_creds_size> creds{};
+					// NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+					const auto ec = irods::base64_decode(
+						reinterpret_cast<unsigned char*>(authorization.data()),
+						authorization.size(),
+						creds.data(),
+						&size);
+					log::debug("{}: base64 - error code=[{}], decoded size=[{}]", fn, ec, size);
+
+					// NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+					std::string_view sv{reinterpret_cast<char*>(creds.data()), size};
+					log::debug("{}: base64 decode credentials = [{}]", fn, sv); // TODO Don't print the password
+
+					const auto colon = sv.find(':');
+					if (colon == std::string_view::npos) {
+						return _sess_ptr->send(fail(status_type::unauthorized));
+					}
+
+					std::string username{sv.substr(0, colon)};
+					std::string password{sv.substr(colon + 1)};
+					log::debug(
+						"{}: username=[{}], password=[{}]", fn, username, password); // TODO Don't print the password
+
+					// BEGIN OG OAUTH THING
+					BodyArguments args{
+						{"client_id",
+					     irods::http::globals::oidc_configuration().at("client_id").get_ref<const std::string&>()},
+						{"grant_type", "password"},
+						{"scope", "openid"},
+						{"username", username},
+						{"password", password}};
+
+					// Query endpoint
+					nlohmann::json res_item{hit_token_endpoint(encode_body(args))};
+
+					// Assume passed, get oidc token
+					const std::string jwt_token{res_item.at("id_token").get<const std::string>()};
+
+					// Feed to JWT parser
+					auto decoded_token{jwt::decode<jwt::traits::nlohmann_json>(jwt_token)};
+
+					// Get irods username
+					// Zones?
```

Comment:
No need to worry about zones. iRODS users can only authenticate against the zone which they were created in AFAIK.